server/shared/publishing/publishing.service.js

```diff
@@ -12,6 +13,10 @@ class PublishingService {
     this.queue = new PromiseQueue(1, Infinity);
   }
 
+  fetchActivity(activity) {
+    return this.queue.add(() => fetchContainer(activity));
+  }
```

Comment:
This has nothing to do with publishing. I understand that it was convenient to expose it through publishing service so you could reference it using publishing service singleton later on but it is definitely wrong move.